upb/decode.c

```diff
@@ -1,33 +1,36 @@
 
+#include "upb/decode.h"
+
 #include <setjmp.h>
 #include <string.h>
 
-#include "upb/decode.h"
 #include "upb/upb.h"
+#include "upb/upb.int.h"
 
+/* Must be last. */
 #include "upb/port_def.inc"
 
 /* Maps descriptor type -> upb field type.  */
```

Comment:
why is this changing? the comment now seems at least out of date. But what causes this to change?

Reply:
Updated the comment. This is changing because I changed the interfaces in `upb/msg.h` to use the `elem_size_lg2` value directly. Before it was:

decoder/gen-code/reflection (elem_type) -> upb/msg.h (elem_type) -> upb/msg.c (maps elem_type -> elem_size_lg2)

It was silly to have the decoder and generated code deal in elem_type and make upb/msg.c do a translation, because they are capable of using elem_size_lg2 directly. So now it's:

decoder/gen-code (elem_size_lg2) -> upb/msg.h (elem_size_lg2) -> upb/msg.c (elem_size_lg2)

and

reflection (maps elem_type -> elem_size_lg2) -> upb/msg.h(elem_size_lg2) -> upb/msg.c (elem_size_lg2)

This moves a table from upb/msg.c to reflection, and speeds up the decoder/gen-code case a tiny bit.